plugins/module_utils/image_policy/params_spec.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2024 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+__author__ = "Allen Robel"
+
+import inspect
+import logging
+from typing import Any, Dict
+
+
+class ParamsSpec:
+    """
+    Parameter specifications for the dcnm_image_policy module.
+    """
+
+    def __init__(self, ansible_module):
+        self.class_name = self.__class__.__name__
+        self.ansible_module = ansible_module
+
+        self.log = logging.getLogger(f"dcnm.{self.class_name}")
+        self.log.debug("ENTERED ParamsSpec()")
+
+        self._params_spec: Dict[str, Any] = {}
+
+    def commit(self):
+        """
+        build the parameter specification based on the state
+        """
+        method_name = inspect.stack()[0][3]
+
+        if self.ansible_module.params["state"] is None:
+            self.ansible_module.fail_json(msg="state is None")
+
+        if self.ansible_module.params["state"] == "merged":
+            # self._build_params_spec_for_merged_state()
```

Comment:
Opps.  This can go in as-is.  But I'd meant to remove "proposed" from the method name since the proposed spec is now the actual spec.  Anyway, I can fix this later and the functionality is as it should be.